src/window.rs

```diff
@@ -295,6 +296,20 @@ impl WindowBuilder {
         platform_impl::Window::new(&window_target.p, self.window, self.platform_specific)
             .map(|window| Window { window })
     }
+
+    /// Builds the window without an event loop.
+    ///
+    /// Intended for use with embedded windows where external code controls the main event loop.
+    /// Embedded windows should be created with `WindowBuilderExtWindows.with_parent_window` or
+    /// other similar platform-specific functions.
+    ///
+    /// Call `Window.set_event_handler` to set an event handler for the window.
+    ///
+    /// Possible causes of error include denied permission, incompatible system, and lack of memory.
+    #[inline]
+    pub fn build_without_event_loop(self) -> Result<Window, OsError> {
```

Comment:
After some pondering, I'm 80% certain this function can not be implemented with X11. Pinging @murarth to investigate this. Surely, no eventsloop = no `XConnection`, right?

Reply:
And while we're at it, no `EventLoop` = no connection to the Wayland display server... I'd be surpized if this is implementable on any of the other platforms.

Reply:
I haven't done any testing and I'm not totally clear on the details of the proposed API (I skimmed through the comments of the above mentioned issues, but I'm not familiar with the use of VST plugins or any other type of GUI plugin), but I think this could be possible.

While it's true that, presently, the `EventLoop` is responsible for opening the `XConnection`, that doesn't necessarily need to be the case.

Reply:
Ok, but what do you think should be responsible for opening it. Keep in mind, we might bump into this, but for X11 if we do something too wierd: https://github.com/rust-windowing/glutin/issues/1152

Reply:
It appears there's already a safeguard against this for X11. All `XConnection` instances are derived from the `lazy_static` here: https://github.com/rust-windowing/winit/blob/master/src/platform_impl/linux/mod.rs#L48

Reply:
Right- but the "external code [that] controls the main event loop" might not be run by winit. So, that external code might be using a different X11 display.

Reply:
True, but that is not an issue unique to winit. Any host program using an X (or Wayland) display server that expects a plugin to create a subwindow to its window must be expected to provide sufficient information to do so. If it doesn't, that would be a problem for any plugin.

The winit-specific element of this would be if, say, a host program provides an Xlib `Display*` that it expects the plugin to use, then we would need something like a `WindowBuilder::from_x_display` method to make use of it.

Reply:
Yes. Which means this function is unfeasible for X11. I suspect for similar reasons it will be unfeasible on every single other platform. This needs to be a platform specific extension like glutin's `RawContextExt`.

Reply:
Oh, right. Yes. I agree. (I forgot that we were talking about this specific platform-independent interface.)

Reply:
> Yes. Which means this function is unfeasible for X11. I suspect for similar reasons it will be unfeasible on every single other platform. This needs to be a platform specific extension like glutin's `RawContextExt`.

This really shouldn't be a platform-specific extension, or rather, if it is there would have to be versions for all major platforms for it to be useful. Embedded/externally hosted windows are definitely possible on x11, windows, and mac. There are VST (and other) plugins for all three.

Looking at https://github.com/steinbergmedia/vstgui/blob/59e9dcfe61e4c4b15c5598ebe7d2e9a52fe6a33d/vstgui/lib/platform/linux/x11platform.cpp it seems that on X11 (but not on Win32 or Cocoa), an own `xcb_connect` call is issued and an event loop is started. I'm not sure how this is done without breaking the existing host loop (I'm no X11 expert). Maybe the `xcb_aux_sync` call is important here, or the host is expected to run separate threads for each plugin window. But they certainly don't on Mac or Win32.

> True, but that is not an issue unique to winit. Any host program using an X (or Wayland) display server that expects a plugin to create a subwindow to its window must be expected to provide sufficient information to do so. If it doesn't, that would be a problem for any plugin.

The APIs I know mostly just provide a single platform-specific pointer/handle/integer to the parent window. But somehow there are plugins on X11 anyways. So it is not unfeasible.